cvmfs/network/download.cc

```diff
@@ -689,6 +701,15 @@ void *DownloadManager::MainDownload(void *data) {
                  "Number of CURL redirects %" PRId64 , info->id(), redir_count);
 
         curl_multi_remove_handle(download_mgr->curl_multi_, easy_handle);
+
+        // let's notify CURL is done and wait for the finishing of
+        // decompressing the data so that VerifyAndFinalize executes correctly
+        if (info->IsValidDataTube()) {
+          DataTubeElement *ele = new DataTubeElement(kActionEndOfData);
+          info->GetDataTubePtr()->EnqueueBack(ele);
+          info->GetDataTubePtr()->Wait();
```

Comment:
It may be necessary to let the other side of the tube signal when it's done. So that we don't need to wait here. For that signalling to work, we would actually need again the `kActionEndOfData`, which could trigger another bit to be send into the `poll` event loop. Let's discuss.